playwright/README.md

```diff
@@ -63,6 +74,23 @@ To open a downloaded CI test report from anywhere in your computer:
 npx playwright show-report name-of-my-extracted-playwright-report
 ```
 
+## Creating tests
+
+Use `test.run()` to create a test. `run` function take an optional object parameter `options` that let you configure how you want to run the test.
+You can configure if you want a logged in test and also an onboarded test.
+Default is not logged in and onboarded.
+
+Example for a logged in user not onboarded:
+
+```ts
+test.run({ loggedIn: true, onboarded: false })("I can ...", 
```

Comment:
Do you see a benefit to preferring functions rather than configuration? For example:

```ts
test("I can...", async () => {
  await logIn()
  // await logIn({ username: "username", password: "password" })

  await onboard()
})
```

With functions, we can script when login/logout occurs, as well as any other actions we may want to orchestrate. In addition, we can configure the actions as needed, such as different usernames and passwords.

It also means we can use the vanilla Playwright API rather than a wrapper.

With a configuration based approach, we could end up in configuration overload with things like: `loggedIn`, `wroomUsername`, `wroomPassword`, `onboardBeforeLogin`, etc.

Just a suggestion since I'm not as familiar with the finer details of how this all works as you are. 🙂

Reply:
I wanted to rely on Fixture functionality from Playwright:  https://playwright.dev/docs/test-fixtures

It gave us an isolated environment to work on. While doing your way is technically possible (I would need to try because you cannot do just `login()`, you need to pass down the browser. Also, I'm not sure because Fixture as I dit let you override the default page object that is used everywhere. 

So the recommendation by playwright is to do it with a Fixture, but then when you want to opt out of the default configuration, you handle it with `use` for example.
Problem with `use` is that it's for a whole file. And it's a bad DX for our usage, having it locally for a specific test is really better and so that why I'm only relying on Fixture with a function to do it.

Last point with fixtures is that you want to prevent mixing your setup inside the test. That's what fixture are for, for example if you need a custom type, you can usage the customType fixture and it will create a customType for you. 
Each test should only handle the test content and we try to prevent having a mix with the config.

At the moment only mocking is kind of allowed because it's also something we try to prevent doing. I will explain in another comment more suited for the mocking part.

I can try using a function inside test but I'm not sure of the outcome, much more with the comment about having an explicit logout for example. It's not the responsibility of the test to call logout if you just want to test an empty state for example.

It's a good point to discuss over a call maybe I would like to continue that point.